### PR TITLE
fix: long directory name error

### PR DESCRIPTION
**Pull request by umerm-work**
### Checklist

- [ ] I have read the [coding guide](https://github.com/ethersphere/bee/blob/master/CODING.md)
- [ ] My change requires a documentation update and I have done it
- [ ] I have added tests to cover my changes.

### Description
fix for POST /bzz - Very long directory name yields store dir err: read tar stream: archive/tar: invalid tar header

### Screenshots (if appropriate):

---
This change is [Reviewable](https://reviewable.io/reviews/ethersphere/bee/3133)

### CODE REVIEW

File: pkg/api/dirs.go [hunk 1]
```diff
@@ -94,6 +95,8 @@ func (s *Service) dirUploadHandler(w http.ResponseWriter, r *http.Request, store
 			jsonhttp.PaymentRequired(w, "batch is overissued")
 		case errors.Is(err, errEmptyDir):
 			jsonhttp.BadRequest(w, errEmptyDir)
+		case err.Error() == errInvalidTar.Error():
```
Comment (notanatol) [inline pkg/api/dirs.go:98]: can this be replaced by `errors.Is(err, errInvalidTar)` ?

**Review by notanatol — APPROVED** [review]

File: pkg/api/dirs.go (hunk 1 above)
Comment (umerm-work) [reply pkg/api/dirs.go:98]: This doesn't work somehow. I tried it.

Comment (aloknerurkar) [reply pkg/api/dirs.go:98]: The `errors.Is` doesnt work as this errInvalidTar is not actually returned from the function. It is a hard-coded string error. This is not the right way to do error checking.

https://pkg.go.dev/archive/tar#pkg-variables

You need to check `errors.Is(err, tar. ErrHeader)`

Comment (aloknerurkar) [reply pkg/api/dirs.go:98]: If you were looking at errEmptyDir, it is actually being returned below(L206). This is why the check works.

**Review by mrekucci — COMMENTED** [review]
> Reviewed 1 of 1 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/ethersphere/bee/3133)* status: all files reviewed, 1 unresolved discussion (waiting on @notanatol)

___
*[`pkg/api/dirs.go` line 35 at r1](https://reviewable.io/reviews/ethersphere/bee/3133#-N90qDne84ulSffnF279:-N90qDne84ulSffnF27A:b-ta2066) ([raw file](https://github.com/ethersphere/bee/blob/35261ec031f3634497365eadda1ab7f84ad2a072/pkg/api/dirs.go#L35)):*
> ```Go
> 
> var errEmptyDir = errors.New("no files in root directory")
> var errInvalidTar = errors.New("read tar stream: archive/tar: invalid tar header")
> ```

The error description should be simple, e.g.: `var errInvalidTarHeader = errors.New("invalid tar header")`. Then if you want to add some context you can wrap the error with `fmt.Errorf("read tar stream: archive/tar: %w", errInvalidTarHeader)`.

File: pkg/api/dirs.go (hunk 1 above)
Comment (umerm-work) [reply pkg/api/dirs.go]: @aloknerurkar Thanks. Now it's better.

**Review by notanatol — APPROVED** [review]

**Review by umerm-work — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/ethersphere/bee/3133)* status: 0 of 1 files reviewed, 1 unresolved discussion (waiting on @mrekucci and @notanatol)

___
*[`pkg/api/dirs.go` line 35 at r1](https://reviewable.io/reviews/ethersphere/bee/3133#-N90qDne84ulSffnF279:-N90t3YRDPU8F-FYzSGx:b-o02ghv) ([raw file](https://github.com/ethersphere/bee/blob/35261ec031f3634497365eadda1ab7f84ad2a072/pkg/api/dirs.go#L35)):*
<details><summary><i>Previously, mrekucci wrote…</i></summary><blockquote>

The error description should be simple, e.g.: `var errInvalidTarHeader = errors.New("invalid tar header")`. Then if you want to add some context you can wrap the error with `fmt.Errorf("read tar stream: archive/tar: %w", errInvalidTarHeader)`.
</blockquote></details>

I have removed it. It is no longer required.

**Review by aloknerurkar — APPROVED** [review]

**Review by mrekucci — APPROVED** [review]
> Reviewed 1 of 1 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/ethersphere/bee/3133)* status: all files reviewed, 1 unresolved discussion (waiting on @notanatol)

**Review by notanatol — APPROVED** [review]

**Pull request merged**